File: frontend/src/models/AuthorOverviewDto.ts
import { logTypeMissmatch, parseNumber, parseString } from '../helpers/model'

export class AuthorOverviewDto {
  authorKey: number = 0
  displayName: string = ''
  platformId: string = ''
  accountId: string = ''
  description: string = ''
  firstPublished: string = ''
  lastPublished: string = ''
  avgBpm: number = 0
  avgDuration: string = ''
  totalSongs: number = 0

  static fromJSON(obj: any): AuthorOverviewDto {
    if (!obj) {
      logTypeMissmatch(`AuthorOverviewDto`, obj)
      return new AuthorOverviewDto()
    }

    return {
      authorKey: parseNumber(obj.authorKey),
      displayName: parseString(obj.displayName),
      platformId: parseString(obj.platformId),
      accountId: parseString(obj.accountId),
      description: parseString(obj.description),
      firstPublished: parseString(obj.firstPublished),
      lastPublished: parseString(obj.lastPublished),
      avgBpm: parseNumber(obj.avgBpm),
      avgDuration: parseString(obj.avgDuration),
      totalSongs: parseNumber(obj.totalSongs),
    }
  }
}

File: ATCD.Backend.Dto/Web/AuthorOverviewDto.cs
namespace ATCD.Backend.Dto.Web
{
    public class AuthorOverviewDto
    {
        public int AuthorKey { get; set; }
        public string DisplayName { get; set; }
        public string PlatformId { get; set; }
        public string AccountId { get; set; }
        public string Description { get; set; }
        public DateTime FirstPublished { get; set; }
        public DateTime LastPublished { get; set; }
        public double AvgBpm { get; set; }
        public string AvgDuration { get; set; }
        public int TotalSongs { get; set; }
    }
}